src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py

```diff
@@ -161,6 +173,15 @@ def validate_arguments(self):
             if environment_name is not None:
                 raise ValidationError(f"Do not pass environment name when using container type {container_type}")
 
+        if self.get_argument_max_alive_period() is not None and self.get_argument_cooldown_period_in_seconds() is not None:
+            raise ValidationError("max_alive_period and cooldown_period cannot be set at the same time.")
+
+        if self.get_argument_max_alive_period() is not None and self.get_argument_lifecycle_type() is not None and self.get_argument_lifecycle_type().lower() != "oncontainerexit":
+            raise ValidationError("max_alive_period can only be set when lifecycle_type is 'OnContainerExit'.")
```

Comment:
can we add some validation that OnContainerExit only supported in custom container pool. Ignore if already validated.